File: scratch/check_dupes.py

import re

with open('/Users/hardikhadwani/Documents/Source Codes/PromptWar/PromptWar_Hardik/assets/js/i18n.js', 'r') as f:
    content = f.read()

# Find all blocks like 'en: { ... }'
blocks = re.findall(r'([a-z]{2}): \{([\s\S]*?)\}', content)

for lang, body in blocks:
    # Find keys like 'key:' at the start of a line (after whitespace)
    keys = re.findall(r'^\s+([a-z0-9_]+):', body, re.MULTILINE)
    seen = set()
    dups = []
    for k in keys:
        if k in seen:
            dups.append(k)
        seen.add(k)
    
    if dups:
        print(f"Duplicates in {lang}: {dups}")
    else:
        print(f"{lang} is clean")

File: scratch/check_dupes.js

const fs = require('fs');
const content = fs.readFileSync('/Users/hardikhadwani/Documents/Source Codes/PromptWar/PromptWar_Hardik/assets/js/i18n.js', 'utf8');

// Use a simple regex to find all keys in each block
const blocks = content.match(/[a-z]{2}: \{[\s\S]*?\}/g);

if (!blocks) {
    console.log("No blocks found");
    process.exit(0);
}

blocks.forEach(block => {
    const lang = block.substring(0, 2);
    const keys = block.match(/^\s+([a-z0-9_]+):/gm).map(k => k.trim().replace(':', ''));
    const seen = new Set();
    const dups = [];
    keys.forEach(k => {
        if (seen.has(k)) dups.push(k);
        seen.add(k);
    });
    if (dups.length > 0) {
        console.log(`Duplicates in ${lang}:`, dups);
    } else {
        console.log(`${lang} is clean`);
    }
});
